ami-build/scripts/install-fun.sh

```diff
@@ -0,0 +1,223 @@
+    jq)
+APT_PKGS_BASE=(
+    apt-transport-https
+    ca-certificates
+    gnupg-agent
+    software-properties-common
+    wget)
+APT_PKGS_BUILD=(
+    docker-ce
+    docker-ce-cli
+    containerd.io
+    expect
+    skopeo
+    zstd)
+AZP_USER=azure-pipelines
+BAZEL_REMOTE_VERSION=2.4.0
+BAZEL_REMOTE_INSTALL_URL="https://github.com/buchgr/bazel-remote/releases/download/v${BAZEL_REMOTE_VERSION}/bazel-remote-${BAZEL_REMOTE_VERSION}-linux-${FULL_ARCH}"
+BAZELISK_VERSION=1.11.0
+BAZELISK_INSTALL_URL="https://github.com/bazelbuild/bazelisk/releases/download/v${BAZELISK_VERSION}/bazelisk-linux-${UNAME_ARCH}"
+export DEBIAN_FRONTEND=noninteractive
+DOCKER_PK=9DC858229FC7DD38854AE2D88D81803C0EBFCD88
+# Hardcoded Github SSH RSA SHA to ensure we download the known key.
+# This must be updated when Github change their public key.
+GITHUB_PK_SHA=uNiVztksCsDhcc0u9e8BujQXVUpKZIDTMczCvj3tD2s
+# https://software.opensuse.org/download/package?package=skopeo&project=devel%3Akubic%3Alibcontainers%3Astable#manualUbuntu
+KUBIC_REPO_URL="https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable/xUbuntu_20.04"
+
+alias apt-get="apt-get -qq"
+
+
+_run_as_azp () {
+    sudo -u "$AZP_USER" /bin/bash -c "$1"
+}
+
+
+agent_install_cleanup () {
+    chown root:root /home/ubuntu/scripts/*.sh
+    mv /home/ubuntu/scripts/run-fun.sh /usr/local/share
+    chmod 0755 /home/ubuntu/scripts/*.sh
+    rm /home/ubuntu/scripts/install-fun.sh
+    mv /home/ubuntu/scripts/*.sh /usr/local/bin
+    rm -rf /home/ubuntu/scripts /home/ubuntu/services
+}
+
+
+apt_add_git () {
+    apt-add-repository -y ppa:git-core/ppa
+}
+
+
+apt_add_docker () {
+    wget -q -O - https://download.docker.com/linux/ubuntu/gpg | apt-key add -
+    apt-key adv --list-public-keys --with-fingerprint --with-colons 0EBFCD88 2>/dev/null \
+        | grep 'fpr' \
+        | head -n1 \
+        | grep "$DOCKER_PK"
+    add-apt-repository -y \
+         "deb [arch=${UNAME_ARCH}] https://download.docker.com/linux/ubuntu \
+         $(lsb_release -cs) \
+         stable"
+}
+
+
+apt_add_skopeo () {
+    echo "${KUBIC_REPO_URL} /" \
+        | tee /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
+    curl -fsSL "${KUBIC_REPO_URL}/Release.key" \
+        | gpg --dearmor \
+        | tee /etc/apt/trusted.gpg.d/devel_kubic_libcontainers_stable.gpg \
+               > /dev/null
+}
+
+
+apt_cleanup () {
+    apt-get autoremove -y --purge
+    rm -rf /var/cache/apt /var/lib/apt/lists/*
+}
+
+
+apt_install_base_pkgs () {
+    if ! apt-get install -y "${APT_PKGS_BASE[@]}"; then
+        echo "Waiting on Ubuntu's crapt setup ..."
+        sleep 5
+        apt-get install -y "${APT_PKGS_BASE[@]}"
+    fi
+}
+
+
+apt_install_pkgs () {
+    echo "Install packages: ${*}"
+    apt-get update
+    apt-get install --no-install-recommends -y "$@"
+}
+
+
+apt_setup () {
+    echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections
+    systemctl stop unattended-upgrades
+    sleep 1
```

Comment:
Can we add a comment explaining the sleep?

Reply:
done